sympy/sets/sets.py

```diff
@@ -1525,6 +1504,21 @@ def as_relational(self, symbol):
 
         return And(A_rel, B_rel)
 
+    @property
+    def is_iterable(self):
+        if self.args[0].is_iterable:
+            return True
```

Comment:
Can this just be `return self.args[0].is_iterable` or is there a need for `is_iterable` to be a fuzzy bool?

Reply:
It would not be determined to be not iterable for cases like 
![image](https://user-images.githubusercontent.com/34944973/64456583-7f3cca00-d12b-11e9-9393-c4996ccc0b5f.png) even if the first argument would return `False`.

Reply:
Do any other sets uses fuzzy logic for is_iterable? It seems to me that iterable is a low-level Python concept rather than a symbolic one and shouldn't need to be None

Reply:
But I don't think that there is any advantage of defining sympy predicates to have no mathematical sense.
`__iter__`, `__contains__`, `__len__` are reserved, but sympy predicates are not reserved by any python methods.
And it is always possible to check `isinstance(A, Iterable)` from collections abc, which should always return `True` or `False`.
And also it is capitalized as `Iterable`

Reply:
I don't see the mathematical meaning of iterable. To me it's just a programmatic construct. Fundamentally `__iter__` will yield elements or will raise and I'd expect `is_iterable` to tell me which. For the 3 `__iter__` methods in this diff it does since all 3 raise when `is_iterable` is not True.

Reply:
Well, the original idea for the `__iter__` was to provide a way to 'try-out' computation for uncertain things, until reaching an undecidable logic.
But I think that in that context, `is_iterable` checks can be totally omitted like.

Now, except for intersection, every set operators iterations are made for some trial computation, and intersection will skip some strictly `is_iterable == False` and will attempt to evaluate even if `is_iterable == None`.

Well, I still have an statement to say that such ternary logic have more advantage, to see if one is totally safe to compute, dangerous to compute, or undecidable. 

And I think that set intersection can further be improved to use such hierarchy, to search for the most safe ones, then try to compute potentially dangerous ones.